src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawAdapter.java

```diff
@@ -26,12 +26,15 @@ public void operateTo(int x, int y) {
 		ILine line = LineFactory.getBasicLine();
 		line.setStartCoordinates(this.startX, this.startY);
 		line.setEndCoordinates(x, y);
+		setPosition(x,y);
 
-		drawLine(line);
+		DrawerFeature.getDrawerController().drawLine(line);
 	}
 
 	@Override
 	public String toString() {
 		return "@Q!$!@$!#@$(*#@&Q(%^*#@";
 	}
 }
+//adapter klasy stosuje się kiedy klasa nie może być użyta jeśli ma niekompatybilny interfejs
+//lub gdy nie ma sie kodu ódłowego klas a nie mozna zmienic ich interfejsu, a jeśli ma się kod zródłowy klasy to nie powinno sie zmieniac interfejsu klasy
```

Comment:
Incorrect answer: the question concerned exclusively the CLASS variant of the adapter pattern.